test/e2e/common.go

```diff
@@ -670,6 +679,125 @@ func runCommand(logFolder, filename, machineIP, user, command string) error {
 	return nil
 }
 
+func WaitForHealthCheckCurrentHealthyToMatch(ctx context.Context, cli client.Client, number int, healthcheck *clusterv1.MachineHealthCheck, timeout, frequency time.Duration) {
+	Eventually(func(g Gomega) int32 {
+		g.Expect(cli.Get(ctx, client.ObjectKeyFromObject(healthcheck), healthcheck)).To(Succeed())
+		return healthcheck.Status.CurrentHealthy
+	}, timeout, frequency).Should(Equal(number))
+}
+
+func WaitForRemediationRequest(ctx context.Context, cli client.Client, healthcheckName types.NamespacedName, toExist bool, timeout, frequency time.Duration) {
+	Eventually(func(g Gomega) {
+		remediation := &infrav1.Metal3Remediation{}
+		if toExist {
+			g.Expect(cli.Get(ctx, healthcheckName, remediation)).To(Succeed())
+		} else {
+			g.Expect(cli.Get(ctx, healthcheckName, remediation)).NotTo(Succeed())
+		}
+	}, timeout, frequency).Should(Succeed())
+}
+
+func DeployControlplaneHealthCheck(ctx context.Context, cli client.Client, namespace, clusterName string) (*clusterv1.MachineHealthCheck, error) {
+	remediationTemplateName := "controlplane-remediation-request"
+	healthCheckName := "controlplane-healthcheck"
+	matchLabels := map[string]string{
+		"cluster.x-k8s.io/control-plane": "",
+	}
+	healthcheck, err := DeployMachineHealthCheck(ctx, cli, namespace, clusterName, remediationTemplateName, healthCheckName, matchLabels)
+	if err != nil {
+		return nil, fmt.Errorf("creating controlplane healthcheck failed: %w", err)
+	}
+	return healthcheck, nil
+}
+
+func DeployWorkerHealthCheck(ctx context.Context, cli client.Client, namespace, clusterName string) (*clusterv1.MachineHealthCheck, error) {
+	remediationTemplateName := "worker-remediation-request"
+	healthCheckName := "worker-healthcheck"
+	matchLabels := map[string]string{
+		"nodepool": "nodepool-0",
+	}
+	healthcheck, err := DeployMachineHealthCheck(ctx, cli, namespace, clusterName, remediationTemplateName, healthCheckName, matchLabels)
+	if err != nil {
+		return nil, fmt.Errorf("creating worker healthcheck failed: %w", err)
+	}
+	return healthcheck, nil
+}
+
+func DeployMachineHealthCheck(ctx context.Context, cli client.Client, namespace, clusterName, remediationTemplateName, healthCheckName string, matchLabels map[string]string) (*clusterv1.MachineHealthCheck, error) {
```

Comment:
For these new functions, could you please add some simple doc string explaining what they do. It doesn't have to be fancy, just something simple like:

```suggestion
// DeployMachineHealthCheck creates a Metal3RemediationTemplate and MachineHealthCheck.
func DeployMachineHealthCheck(ctx context.Context, cli client.Client, namespace, clusterName, remediationTemplateName, healthCheckName string, matchLabels map[string]string) (*clusterv1.MachineHealthCheck, error) {
```

I would probably rename the `Deploy*` functions to `Create*` but that is more a suggestion. It is just that they use `cli.Create` and we already have many other functions that `create` stuff, like `createTargetCluster`.